x-pack/plugins/enterprise_search/public/applications/app_search/components/log_retention/messaging/constants.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { FormattedDate, FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+
+import { LogRetentionOptions } from '../types';
+
+export const renderLogRetentionDate = (dateString: string) => (
+  <FormattedDate value={new Date(dateString)} month="long" day="numeric" year="numeric" />
+);
+
+const CAPITALIZATION_MAP = {
+  [LogRetentionOptions.Analytics]: {
+    capitalized: i18n.translate(
+      'xpack.enterpriseSearch.appSearch.logRetention.type.analytics.title.capitalized',
+      { defaultMessage: 'Analytics' }
+    ),
+    lowercase: i18n.translate(
+      'xpack.enterpriseSearch.appSearch.logRetention.type.analytics.title.lowercase',
+      { defaultMessage: 'analytics' }
+    ),
+  },
+  [LogRetentionOptions.API]: {
+    capitalized: i18n.translate(
+      'xpack.enterpriseSearch.appSearch.logRetention.type.api.title.capitalized',
+      { defaultMessage: 'API' }
+    ),
+    lowercase: i18n.translate(
+      'xpack.enterpriseSearch.appSearch.logRetention.type.api.title.lowercase',
+      { defaultMessage: 'API' }
+    ),
+  },
+};
+
+interface Props {
+  type: LogRetentionOptions;
+  disabledAt?: string | null;
+  minAgeDays?: number | null;
```

Comment:
Raises an eyebrow to me that these can be both undefined and are nullable, two different ways of representing when a value is "missing".  If these represent distinct states, we might be doing too much.

Reply:
🤷 That's the current definition in ent-search, I think we just copied it over. Looks like you originally defined this 6 months ago there? I have no strong feelings either way, this comes in from the server so it would basically just be confirming that the server can or cannot serve a null

Reply:
Oh if you're raising your eyebrow at the undefined / `?`, I mean that's because we're passing around optional props. That's probably easily solvable by doing `minAgeDays?: LogRetentionPolicy[minAgeDays]`. I can definitely make that change.

Reply:
390ace7